### PR TITLE
Add terminationGracePeriodSeconds

### PR DESCRIPTION
**Pull request by vostres**

### CODE REVIEW

File: charts/streams-app/templates/deployment.yaml [hunk 1]
```diff
@@ -48,7 +48,7 @@ spec:
         prometheus.io/scrape: "true"
         prometheus.io/port: {{ .Values.prometheus.jmx.port | quote }}
       {{- end }}
-      {{- range $key, $value := .Values.annotations }}
+      {{- range $key, $value := .Values.podAnnotations }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:51]: Please add a default empty list to values.yaml

File: charts/streams-app/templates/deployment.yaml [hunk 2]
```diff
@@ -58,7 +58,7 @@ spec:
       labels:
         app: {{ template "streams-app.name" . }}
         release: {{ .Release.Name }}
-        {{- range $key, $value := .Values.labels }}
+        {{- range $key, $value := .Values.podLabels }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:61]: Please add a default empty list to values.yaml

File: charts/streams-app/templates/deployment.yaml [hunk 3]
```diff
@@ -99,6 +99,7 @@ spec:
       {{- if .Values.priorityClassName }}
       priorityClassName: {{ .Values.priorityClassName }}
       {{- end }}
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:102]: Please add a default of 300 to values.yaml

File: charts/streams-app/templates/deployment.yaml (hunk 1 above)
Comment (philipp94831) [reply charts/streams-app/templates/deployment.yaml:51]: There is also a condition above where you need to check podAnnotations instead

File: charts/streams-app/templates/deployment.yaml (hunk 2 above)
Comment (philipp94831) [reply charts/streams-app/templates/deployment.yaml:61]: There is also a condition above where you need to check podLabels instead

**Review by philipp94831 — CHANGES_REQUESTED** [review]

**Comment by philipp94831** [comment]
Please merge master

**Comment by philipp94831** [comment]
Please add podAnnotations/podLabels also to producer-app, producer-app-clean-up-job, streams-app-clean-up-job

File: charts/streams-app/templates/deployment.yaml (hunk 2 above)
Comment (vostres) [reply charts/streams-app/templates/deployment.yaml:61]: > There is also a condition above where you need to check podLabels instead

I don't think there is a condition for Values.labels

Comment (philipp94831) [reply charts/streams-app/templates/deployment.yaml:61]: Sorry, you are right

**Comment by vostres** [comment]
> Please add podAnnotations/podLabels also to producer-app, producer-app-clean-up-job, streams-app-clean-up-job

so in case there are already Values.annotations/Values.labels in chart under `spec.template.metadata` I should just refactor those and if there are none (as in producer-app) add them?

**Comment by philipp94831** [comment]
Yeah, would be good to have proper annotation son producer app jobs/cronjobs

**Comment by philipp94831** [comment]
We should split the PR in two: one for terminationGracePeriod, one for labels/annotations

**Comment by vostres** [comment]
> We should split the PR in two: one for terminationGracePeriod, one for labels/annotations

I guess I than close this pull request and create two new ones?

**Comment by philipp94831** [comment]
> > We should split the PR in two: one for terminationGracePeriod, one for labels/annotations
> 
> I guess I than close this pull request and create two new ones?

Just open one new one. You can use this one for one of the two things

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please update README of the chart

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**